### PR TITLE
Do not set bindToPort for unix domain sockets

### PR DESCRIPTION
**Pull request by rshriram**
If set, this will cause unix domain sockets to not work.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -932,15 +932,20 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(node *model.
...
+				// for older proxies, it defaults to iptables redirect.  If the
+				// listener's capture mode specifies NONE, then the proxy wants
+				// this listener alone to be on a physical port. If the
+				// listener's capture mode is default, then its same as
+				// iptables i.e. bindToPort is false.
+				bindToPort = true
+			} else if strings.HasPrefix(egressListener.IstioListener.Bind, model.UnixAddressPrefix) {
+				// If the bind is a Unix domain socket, set bindtoPort to false as it makes no
+				// sense to have ORIG_DST listener for unix domain socket listeners.
+				bindToPort = true
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener.go:947]: The comment conflicts with the value

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 2]
```diff
@@ -932,15 +932,20 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(node *model.
...
+		} else if egressListener.IstioListener != nil {
+			if egressListener.IstioListener.CaptureMode == networking.CaptureMode_NONE {
+				// proxy uses iptables redirect or tproxy. IF mode is not set
+				// for older proxies, it defaults to iptables redirect.  If the
+				// listener's capture mode specifies NONE, then the proxy wants
+				// this listener alone to be on a physical port. If the
+				// listener's capture mode is default, then its same as
+				// iptables i.e. bindToPort is false.
+				bindToPort = true
+			} else if strings.HasPrefix(egressListener.IstioListener.Bind, model.UnixAddressPrefix) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener.go:944]: I think you want `if egressListener.IstioListener.CaptureMode == networking.CaptureMode_NONE || strings.HasPrefix(egressListener.IstioListener.Bind, model.UnixAddressPrefix)`, right?

Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/listener.go:944]: Yes but since the comments are big, I thought I would split them up into two separate else-ifs

**Comment by rshriram** [comment]
ping @hzxuzhonghu

**Pull request merged**